crates/oro-manifest/src/lib.rs

```diff
@@ -0,0 +1,324 @@
+                url: url.clone(),
+            }),
+            PersonField::Str(s) => parser::parse_person(s.trim()),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Person {
+    pub name: Option<String>,
+    pub email: Option<String>,
+    pub url: Option<String>,
+}
+
+impl FromStr for Person {
+    type Err = Error;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        parser::parse_person(s.trim())
+    }
+}
+
+#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
+pub struct Directories {
+    pub bin: Option<String>,
+    pub man: Option<String>,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Bin {
+    Str(String),
+    Hash(HashMap<String, String>),
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Man {
+    Str(String),
+    Vec(Vec<String>),
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Exports {
+    Str(String),
+    Vec(Vec<String>),
+    Obj(HashMap<String, Exports>),
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Imports {
+    Str(String),
+    Vec(Vec<String>),
+    Obj(HashMap<String, Imports>),
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Repository {
+    Str(String),
+    Obj {
+        #[serde(rename = "type")]
+        repo_type: Option<String>,
+        url: Option<String>,
+        directory: Option<String>,
+    },
+}
+
+mod parser {
+    use super::*;
+
+    use nom::bytes::complete::{take_till, take_till1};
+    use nom::character::complete::char;
+    use nom::combinator::{all_consuming, map, opt};
+    use nom::error::{context, convert_error, ParseError, VerboseError};
+    use nom::sequence::{delimited, tuple};
+    use nom::{Err, IResult};
+
+    pub fn parse_person<I: AsRef<str>>(input: I) -> Result<Person> {
+        let input = &input.as_ref()[..];
+        match all_consuming(person::<VerboseError<&str>>)(input) {
+            Ok((_, arg)) => Ok(arg),
+            Err(err) => Err(Error::ParsePersonError {
+                input: input.into(),
+                msg: match err {
+                    Err::Error(e) => convert_error(input, e),
+                    Err::Failure(e) => convert_error(input, e),
+                    Err::Incomplete(_) => "More data was needed".into(),
+                },
+            }),
+        }
+    }
+
+    fn person<'a, E>(input: &'a str) -> IResult<&'a str, Person, E>
+    where
+        E: ParseError<&'a str>,
+    {
+        context(
+            "person",
+            map(
+                tuple((
+                    opt(take_till1(|c| c == '<')),
+                    opt(delimited(char('<'), take_till1(|c| c == '>'), char('>'))),
+                    take_till(|c| c == '('),
+                    opt(delimited(char('('), take_till1(|c| c == ')'), char(')'))),
```

Comment:
Using `preceded` here would let us skip the `_` arg in the handler below, if that's helpful:

```rust
opt(
    preceded(
        take_till('('),
        delimited(char('('), take_till1(|c| c == ')'), char(')'))
    )
)
```